spec/compiler/semantic/annotation_spec.cr

```diff
@@ -918,6 +918,23 @@ describe "Semantic: annotation" do
     end
   end
 
+  it "errors when annotate instance variable in subclass" do
+    assert_error %(
+      annotation Foo
+      end
+
+      class Base
+        @x : Nil
+      end
+
+      class Child < Base
+        @[Foo]
+        @x : Nil
+      end
+      ),
+      "can't annotate an instance variable not defined in this class"
```

Comment:
Is the message right? Shoudn't it be: `can't annotate an instance variable defined in this class` ?
(as in without the _not_?)

Reply:
The point is that it's defined in `Base` and only *re*defined in `Child` and that's exactly what we want to forbid annotations on. Annotating it in `Base` is ok and annotating it in `Child` if *not* defined in `Base` is ok.

Reply:
But yeah, the double negation  might be a bit hard to follow.  Maybe

>  can only annotate instance variables defined in this class

or even

> cannot annotate @x here because it was first defined in Base

Reply:
Done. More detailed and precise error is always better. Thanks!